Add 'requirements' subsection to active storage guide [ci-skip] (#42754)
rails/rails: guides/source/active_storage_overview.md

representations of non-image uploads like PDFs and videos, and extract metadata from 
arbitrary files.

### Requirements

> Various features of Active Storage depend on third party software which Rails 

Reviewer: Sorry I missed this the first time around! :pray: 

```suggestion
Various features of Active Storage depend on third-party software which Rails 
```

representations of non-image uploads like PDFs and videos, and extract metadata from 
arbitrary files.

### Requirements

> Various features of Active Storage depend on third party applications which Rails 

Reviewer: What about "software", since that is the term used in the warning further below?

```suggestion
Various features of Active Storage depend on third party software which Rails 
```

will not install, and must be included separately.

* [ImageMagick](https://imagemagick.org/index.php) or [libvips](https://github.com/libvips/libvips) v8.6+ 

> for image analysis and transformations;

Author: That line break was accidental 😅

* [ffmpeg](http://ffmpeg.org/) v3.4+ for video/audio analysis, and video previews;
* [poppler](https://poppler.freedesktop.org/) or [muPDF](https://mupdf.com/) for PDF previews;

To enable image transformations and analysis the `image_processing` gem is also necessary:

```ruby
gem 'image_processing', '>= 1.2'
```

> TIP: ImageMagick is the better known and more widely available of the two applications. However, libvips [is known](https://github.com/libvips/libvips/wiki/Speed-and-memory-use) to be up to 10x faster and consume 1/10 the memory. For JPEG files this can be further improved by removing `libjpeg-dev` and installing `libjpeg-turbo-dev` in its place. The latter is 2-7x faster then the former. 

Reviewer: ```suggestion
TIP: Compared to libvips, ImageMagick is better known and more widely available. However, libvips can be [up to 10x faster and consume 1/10 the memory](https://github.com/libvips/libvips/wiki/Speed-and-memory-use). For JPEG files, this can be further improved by replacing `libjpeg-dev` with `libjpeg-turbo-dev`, which is [2-7x faster](https://libjpeg-turbo.org/About/Performance). 
```

Author: @ghiculescu Yes, the first half could be removed if it does (the `libjepg-turbo` part is still useful). I was not sure if it would however. Should I remove it so the PR can be merged or do we wait to see what will happend to #42744 before merging this one?

Reviewer: I think we can go with what we have, and then rebase #42744 to remove any documentation that's no longer applicable.

> WARNING: Before you install and use third-party software, make sure you understand the licensing implications of doing so. MuPDF specifically is licensed under AGPL and requires a commercial license for some use.

Reviewer: ```suggestion
WARNING: Before you install and use third-party software, make sure you understand the licensing implications of doing so. MuPDF, in particular, is licensed under AGPL and requires a commercial license for some use.
```

To add support for another format, add your own previewer. See the
[`ActiveStorage::Preview`][] documentation for more information.

WARNING: Extracting previews requires third-party applications: FFmpeg v3.4+ for

> video and muPDF for PDFs, and on macOS also XQuartz and Poppler.

Reviewer: I do not use macOS...  Is XQuartz no longer required?  Or can we just assume it's installed already?

Author: Oh yes, I forgot to add a comment about this one. XQuartz was not in the original list of dependencies for AST. It was added in 0af43ac7f6583afd3400c176fde88b96af58dc77. I have no idea why. The homebrew formulas for installing ffmpeg, mupdf and poppler do not have it.

Author: I've also been using both ffmpeg and poppler constantly in my mac and they never complained about XQuartz.

Reviewer: Hmm, indeed both ["mupdf" "xquartz"](https://www.google.com/search?q=%22mupdf%22+%22xquartz%22) and ["poppler" "xquartz"](https://www.google.com/search?q=%22poppler%22+%22xquartz%22) yield very few results.  If XQuartz is a transitive dependency of either MuPDF or Poppler, perhaps we can assume that it will be taken care of as part of properly installing those tools.
